src/HandlerResolver.php

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types=1);
+
+namespace Bref\SymfonyBridge;
+
+use Bref\Runtime\FileHandlerLocator;
+use Bref\SymfonyBridge\Http\KernelAdapter;
+use Exception;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+/**
+ * This class resolves handlers.
+ *
+ * For example, if we configure `handler: xyz` in serverless.yml, then Bref
+ * will call this class to resolve `xyz` into the real Lambda handler.
+ */
+class HandlerResolver implements ContainerInterface
+{
+    private ?ContainerInterface $symfonyContainer;
+    private FileHandlerLocator $fileLocator;
+
+    public function __construct()
+    {
+        // Bref's default handler resolver
+        $this->fileLocator = new FileHandlerLocator;
+        $this->symfonyContainer = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get($id)
+    {
+        // By default we check if the handler is a file name (classic Bref behavior)
+        if ($this->fileLocator->has($id)) {
+            return $this->fileLocator->get($id);
+        }
+
+        // If not, we try to get the handler from the Symfony container
+        $handler = $this->symfonyContainer()->get($id);
+
+        // If the kernel was configured as a handler, then we wrap it to make it a valid HTTP handler for Lambda
+        if ($handler instanceof HttpKernelInterface) {
+            $handler = new KernelAdapter($handler);
+        }
+
+        return $handler;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function has($id): bool
+    {
+        return $this->fileLocator->has($id) || $this->symfonyContainer()->has($id);
+    }
+
+    /**
+     * Create and return the Symfony container.
+     */
+    private function symfonyContainer(): ContainerInterface
```

Comment:
What if I want to get a container but not from a Symfony Kernel? 
Or is this exclusively a Symfony solution?

Reply:
I guess in this case you would use the default Bref PSR-11 support and not bother with this bridge ?

https://github.com/brefphp/bref/blob/master/src/Bref.php#L27

Reply:
Yes this is 100% about Symfony (since this is in the symfony bridge project)